docs/topics/classes.md

```diff
@@ -1,245 +1,692 @@
+> On the JVM, if all primary constructor parameters have default values, the compiler implicitly provides 
+> a parameterless constructor that uses those default values. 
+> 
+> This makes it easier to use Kotlin with libraries such as [Jackson](https://github.com/FasterXML/jackson) 
+> or [Spring Data JPA](https://spring.io/projects/spring-data-jpa), which create class objects through 
+> parameterless constructors.
 >
+> In the following example, Kotlin implicitly provides a parameterless constructor `Person()` that uses the default value `""`:
+> 
 > ```kotlin
-> class Customer(val customerName: String = "")
+> class Person(val personName: String = "")
 > ```
 >
 {style="note"}
 
-## Creating instances of classes
+## Creating objects of classes
+
+Like in other object-oriented programming languages, a class in Kotlin serves as a blueprint or template 
+that defines the properties (attributes) and behaviors (functions) of objects. When you create an instance 
+of a class, you are creating a concrete object based on that blueprint.
 
-To create an instance of a class, call the constructor as if it were a regular function. You can assign the created instance to a [variable](basic-syntax.md#variables):
+To create an object of a class, use the class name followed by parentheses (`()`), similar to calling a function:
 
 ```kotlin
-val invoice = Invoice()
+// Creates an object of the 'Person' class
+val person1 = Person()
+```
 
-val customer = Customer("Joe Smith")
+In Kotlin, you can create objects:
+
+* **Without arguments** (`()`): creates an object using a constructor with default values.
+* **With arguments** (`("value")`): creates an object by passing specific values to the constructor.
+
+As you can see, you can assign the created object to a mutable (`var`) or immutable (`val`) [variable](basic-syntax.md#variables):
+
+```kotlin
+// Creates an object by passing a specific value and assigns it to a mutable variable
+var person2 = Person("Joe")
+
+// Creates an object using the default constructor's value and assigns it to an immutable variable
+val person1 = Person()
 ```
 
-> Kotlin does not have a `new` keyword.
->
-{style="note"}
+You can create objects wherever you need them, inside the `main()` function, within other functions, or as part of a class. 
 
-The process of creating instances of nested, inner, and anonymous inner classes is described in [Nested classes](nested-classes.md).
+When creating objects inside the `main()` function, when the program runs, execution starts from `main()`. 
+Here is where you instantiate objects to interact with them. When creating objects of a class, you're invoking constructors 
+(for example, `Person()` or `Person("Joe")`) directly inside the `main()` function.
```

Comment:
I feel that, for some reason, here we begin talking about smth unrelated to classes, i.e., about the purposes of the `main` function, when it is invoked, etc. Maybe we should create a separate page about it then, and simply reference it here?